vpr/src/device/rr_graph_obj.h

```diff
@@ -0,0 +1,647 @@
+     *
+     *             TOP
+     *        +-----------+
+     *        |           |
+     *  LEFT  |   Logic   | RIGHT
+     *        |   Block   |
+     *        |           |
+     *        +-----------+
+     *            BOTTOM
+     *
+     */
+    e_side node_side(const RRNodeId& node) const;
+
+    /* Get resistance of a node, used to built RC tree for timing analysis */
+    float node_R(const RRNodeId& node) const;
+
+    /* Get capacitance of a node, used to built RC tree for timing analysis */
+    float node_C(const RRNodeId& node) const;
+
+    /* Get segment id of a node, containing the information of the routing
+     * segment that the node represents. See more details in the data structure t_segment_inf
+     */
+    RRSegmentId node_segment(const RRNodeId& node) const;
+
+    /* Get the number of non-configurable incoming edges to a node */
+    short node_num_configurable_in_edges(const RRNodeId& node) const;
+    /* Get the number of non-configurable outgoing edges from a node */
+    short node_num_non_configurable_in_edges(const RRNodeId& node) const;
+    /* Get the number of configurable output edges of a node */
+    short node_num_configurable_out_edges(const RRNodeId& node) const;
+    /* Get the number of non-configurable output edges of a node */
+    short node_num_non_configurable_out_edges(const RRNodeId& node) const;
+
+    /* Get the range (list) of edges related to a given node */
+    edge_range node_configurable_in_edges(const RRNodeId& node) const;
+    edge_range node_non_configurable_in_edges(const RRNodeId& node) const;
+    edge_range node_configurable_out_edges(const RRNodeId& node) const;
+    edge_range node_non_configurable_out_edges(const RRNodeId& node) const;
+
+    /* Get a list of edge ids, which are incoming edges to a node */
+    edge_range node_in_edges(const RRNodeId& node) const;
+    /* Get a list of edge ids, which are outgoing edges from a node */
+    edge_range node_out_edges(const RRNodeId& node) const;
+
+    /* Edge-related attributes 
+     * An example to explain the terminology used in RRGraph
+     *          edgeA
+     *  nodeA --------> nodeB
+     *        | edgeB
+     *        +-------> nodeC
+     *
+     *  +----------+----------------+----------------+
+     *  |  Edge Id |  edge_src_node | edge_sink_node |
+     *  +----------+----------------+----------------+
+     *  |   edgeA  |     nodeA      |      nodeB     |
+     *  +----------+----------------+----------------+
+     *  |   edgeB  |     nodeA      |      nodeC     |
+     *  +----------+----------------+----------------+
+     *
+     */
+    size_t edge_index(const RREdgeId& edge) const; /* TODO: deprecate this accessor as outside functions should use RREdgeId */
+    /* Get the source node which drives a edge */
+    RRNodeId edge_src_node(const RREdgeId& edge) const;
+    /* Get the sink node which a edge ends to */
+    RRNodeId edge_sink_node(const RREdgeId& edge) const;
```

Comment:
Suggest a little more vertical space (blank line before the comment block that goes with the next function) so it's clear the comments apply to the function after, not the function before.